packages/toolpad-app/src/toolpad/AppEditor/AppEditorShell.tsx

```diff
@@ -159,14 +164,48 @@ export default function AppEditorShell({ appId, ...props }: ToolpadShellProps) {
           >
             Preview
           </Button>
-          <Button
-            variant="outlined"
-            endIcon={<RocketLaunchIcon />}
-            color="primary"
-            onClick={handleCreateReleaseDialogOpen}
-          >
-            Deploy
-          </Button>
+          <ButtonGroup>
+            <Button
+              variant="outlined"
+              endIcon={!isDeployed ? <RocketLaunchIcon /> : null}
+              size="small"
+              color="primary"
+              onClick={handleCreateReleaseDialogOpen}
+              sx={
+                isDeployed
+                  ? {
+                      // Workaround for https://github.com/mui/material-ui/issues/31210
+                      borderTopRightRadius: 0,
+                      borderBottomRightRadius: 0,
+                    }
+                  : {}
+              }
+            >
+              Deploy
+            </Button>
+            {isDeployed ? (
+              <Button
+                size="small"
+                component="a"
+                href={`/deploy/${appId}`}
+                target="_blank"
+                sx={
+                  isDeployed
+                    ? {
+                        // Workaround for https://github.com/mui/material-ui/issues/31210
+                        borderTopLeftRadius: 0,
+                        borderBottomLeftRadius: 0,
+                        marginLeft: '-1px',
+                      }
+                    : {}
+                }
+              >
+                <Tooltip title="Preview latest deployed version">
```

Comment:
```suggestion
                <Tooltip title="open last deployed version">
```

What we currently call "preview" in Toolpad nomenclature is the version that you're currently editing, i.e. the button left to the deploy button. I'd not conflate the term "preview" with deployments.